File: Stack/stack.js
// Pseudo-Classical pattern

var Stack = function(){
  this.storage = {};
  this.stackSize = 0;
};


Stack.prototype.push = function(value){
  this.storage[this.stackSize++] = value;
};

Stack.prototype.pop = function(){
  if (this.stackSize) {
    var temp = this.storage[this.stackSize - 1];
    return (delete this.storage[this.stackSize--]) && temp;
  }
};

Stack.prototype.size = function(){
  return this.stackSize;
};

File: Stack/stack.rb
class Stack
  def initialize
    @storage = {}
    @stackSize = 0
  end

  def push(value)
  end

  def pop
    if @stackSize > 0
      temp = @storage[@stackSize - 1]
      @storage.delete(@stackSize -= 1);
      return temp
    end
  end

  def size
    return @stackSize
  end

end
